### ISSUE TITLE
mloginfo --queries gives `TypeError: '<' not supported between instances of 'str' and 'int'`

### ISSUE THREAD

**Issue by niccottrell** [rueckstiess/mtools#697]
### Actual/current behavior

```
mloginfo --distinct mongodb.log
     source: mongodb.log
       host: cluster1-shard-00-01-gfskh.mongodb.net:27017
      start: 2019 Jun 23 18:33:11.331
        end: 2019 Jun 24 14:20:56.451
date format: iso8601-local
     length: 181723
     binary: mongod
    version: 4.0.10
    storage: wiredTiger

QUERIES
                                                                                          
Traceback (most recent call last):
  File "/usr/local/bin/mloginfo", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/mtools/mloginfo/mloginfo.py", line 95, in main
    tool.run()
  File "/usr/local/lib/python3.6/site-packages/mtools/mloginfo/mloginfo.py", line 90, in run
    section.run()
  File "/usr/local/lib/python3.6/site-packages/mtools/mloginfo/sections/query_section.py", line 136, in run
    reverse=reverse)
TypeError: '<' not supported between instances of 'str' and 'int'
```
### Environment

| Software         | Version
| ---------------- | -------
| mtools           | 1.5.3 
| Python | 3.6.5 (default, Jun 17 2018, 12:13:06) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
| MongoDB server   |  4.0.9
| Operating system | macOS 14

**Comment by niccottrell** [comment]
Seems to be a sorting problem:

```
table_rows = sorted(table_rows,
                            key=itemgetter(self.mloginfo.args['sort']),
                            reverse=reverse)
```

**Comment by stennie** [comment]
@niccottrell Can you provide a snippet of log to repro the issue? I tried a few random log samples with the same versions of `mloginfo` & Python but didn't encounter this error yet.

Thanks,
Stennie

**Comment by niccottrell** [comment]
Hi Stennie!
I didn’t want to share customer logs on GitHub, but I’ve managed to narrow it down to these 5k lines…Can you reproduce with that?

**Comment by Jagman13** [comment]
Hi I also encountered same issue for one of the logs file. 

Is there any work around for this ?

**Comment by stennie** [comment]
This issue is caused by log lines that are unexpectedly truncated but partially parseable. `mloginfo` can extract a query pattern but uses `-` as a placeholder for aggregated totals based on duration. This leads to an error in Python 3 which doesn't support sorting mixed output types (numeric durations vs string placeholders).

I committed a fix for mtools 1.6.0, but a manual workaround for this error would be to filter your logs to lines that appear to end in a duration:

    grep "\dms$" mongod.log > dur.log && mloginfo --queries dur.log
